components/sceneutil/statesetupdater.hpp

```diff
@@ -34,8 +34,6 @@ namespace SceneUtil
         StateSetUpdater();
         StateSetUpdater(const StateSetUpdater& copy, const osg::CopyOp& copyop);
 
-        META_Object(SceneUtil, StateSetUpdater)
```

Comment:
@glassmancody I think we will want to apply such changes to our upcoming `AutoDepth` so that its details can not leak into serialisation. We will just have to override `clone()` and `cloneType()` that are normally provided by `META` if we want to be able to clone `AutoDepth` instances as `AutoDepth`.

Reply:
I've setup the `AutoDepth` like so and not added to this blacklist, as we need it serialized as an `osg::Depth` (e.g. for generating offline simplified meshes).
```
template <class Cls>
static osg::Object* createInstanceFunc() { return new Cls; }
//...
osgDB::ObjectWrapperManager* mgr = osgDB::Registry::instance()->getObjectWrapperManager();
// ...
mgr->addWrapper(new osgDB::ObjectWrapper(createInstanceFunc<osg::Depth>, "SceneUtil::AutoDepth", "osg::Object osg::StateAttribute osg::Depth"));
```

Reply:
Yes, but I do not understand why you have written a new serialiser that breaks `osg` format compatibility for trivial features. Removing `META`, `className()`, `libraryName()` usage in `AutoDepth` will allow us to inherit the default `osg::Depth` serialiser automatically and avoid breaking its compatibility.

Reply:
I am not familiar with the serializer at all really, the following is sufficient then?

```diff
-        META_StateAttribute(SceneUtil, AutoDepth, osg::StateAttribute::DEPTH);
+
+        osg::Object* cloneType() const override { return new AutoDepth; }
+        osg::Object* clone(const osg::CopyOp& copyop) const override { return new AutoDepth(*this,copyop); }
```

Reply:
Yes, I think that will be sufficient.